src/main/java/com/minecolonies/coremod/colony/buildings/AbstractCitizenAssignable.java

```diff
@@ -0,0 +1,255 @@
+    {
+        // Can be overridden by other buildings.
+    }
+
+    /**
+     * On tick of the world.
+     *
+     * @param event {@link net.minecraftforge.fml.common.gameevent.TickEvent.WorldTickEvent}
+     */
+    public void onWorldTick(final TickEvent.WorldTickEvent event)
+    {
+        // Can be overridden by other buildings.
+    }
+
+    //------------------------- Starting Assigned Citizen handling -------------------------//
+
+    /**
+     * Get the main worker of the building (the first in the list).
+     *
+     * @return the matching CitizenData.
+     */
+    public CitizenData getMainCitizen()
+    {
+        if (assignedCitizen.isEmpty())
+        {
+            return null;
+        }
+        return assignedCitizen.get(0);
+    }
+
+    /**
+     * Returns the worker of the current building.
+     *
+     * @return {@link CitizenData} of the current building
+     */
+    public List<CitizenData> getAssignedCitizen()
+    {
+        return new ArrayList<>(assignedCitizen);
+    }
+
+    /**
+     * Method to remove a citizen.
+     *
+     * @param citizen Citizen to be removed.
+     */
+    public void removeCitizen(final CitizenData citizen)
+    {
+        if (isCitizenAssigned(citizen))
+        {
+            assignedCitizen.remove(citizen);
+            citizen.setWorkBuilding(null);
+            markDirty();
+        }
+    }
+
+    /**
+     * Returns if the {@link CitizenData} is the same as the worker.
+     *
+     * @param citizen {@link CitizenData} you want to compare
+     * @return true if same citizen, otherwise false
+     */
+    public boolean isCitizenAssigned(final CitizenData citizen)
+    {
+        return assignedCitizen.contains(citizen);
+    }
+
+    /**
+     * Returns the first worker in the list.
+     *
+     * @return the EntityCitizen of that worker.
+     */
+    public Optional<EntityCitizen> getMainCitizenEntity()
+    {
+        if (assignedCitizen.isEmpty())
+        {
+            return Optional.empty();
+        }
+        return assignedCitizen.get(0).getCitizenEntity();
+    }
+
+    /**
+     * Returns whether or not the building has a worker.
+     *
+     * @return true if building has worker, otherwise false.
+     */
+    public boolean hasAssignedCitizen()
+    {
+        return !assignedCitizen.isEmpty();
+    }
+
+    /**
+     * Returns the {@link net.minecraft.entity.Entity} of the worker.
+     *
+     * @return {@link net.minecraft.entity.Entity} of the worker
+     */
+    @Nullable
+    public List<Optional<EntityCitizen>> getAssignedEntities()
+    {
+        return assignedCitizen.stream().filter(Objects::nonNull).map(CitizenData::getCitizenEntity).collect(Collectors.toList());
+    }
+
+    /**
+     * Assign the citizen to the current building.
+     *
+     * @param citizen {@link CitizenData} of the worker
+     */
+    public boolean assignCitizen(final CitizenData citizen)
+    {
+        if (assignedCitizen.contains(citizen))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)